### ISSUE TITLE
410.8.20190412.1 rhcos release doesn't include the openstack image

### ISSUE THREAD

**Issue by mcornea** [openshift-metalkube/dev-scripts#347]
```shell
curl -s https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/builds.json | jq -r ".builds[0]"
410.8.20190412.1


curl -s https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/410.8.20190412.1/meta.json | jq -r .images
{
  "qemu": {
    "path": "rhcos-410.8.20190412.1-qemu.qcow2",
    "sha256": "b2b80605be197d988816df347cfe76c652db556fc47830df7c74c66e2ac97d28",
    "size": "765151729",
    "uncompressed-sha256": "1a9394ed8383cebb41228e49e4b3b009d7e6a6cc0045061ce0549d3ed860ccc0",
    "uncompressed-size": "2165243904"
  }
}

curl -s https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/410.8.20190412.1/meta.json | jq -r .images.openstack
null

```

Affected code: https://github.com/openshift-metalkube/dev-scripts/blob/master/common.sh#L54

**Comment by gklein** [comment]
Rolling back to an earlier build will workaround this broken build.

In: 

https://github.com/openshift-metalkube/dev-scripts/blob/ecf60de2ea62593bb1613c5ee1af2a888ed3d261/common.sh#L52

Change line 52 to:

export RHCOS_LATEST="$(curl ${RHCOS_IMAGE_URL}/builds.json | jq -r ".builds[**1**]")"

**Comment by hardys** [comment]
Looking at https://github.com/openshift/installer/commit/8599867bef65c539e7fe2ea39dd1d1c5b4df79f5 it seems openshift/installer moved to a new location, and it seems like authentication is needed to access, currently looking at how we can align with that.

**Comment by hardys** [comment]
Hmm https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/builds.json is still working, @gklein do you know if this is a one-off broken build so we can temporarily pin as a workaround, or do we need to be moving to the new `https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/` location?

**Comment by gklein** [comment]
@hardys I don't know what is the best resource, but given my check,
**2 broken builds** in https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa
**0 broken builds** in https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/ 

So I guess https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/  is a safer option.

# cat checkBuilds.sh
#!/bin/bash

IMAGE_URL_PREFIX=$1
BUILDS_JSON="builds.json"
BUILDS="$(curl -s ${IMAGE_URL_PREFIX}/${BUILDS_JSON} | jq -r '.builds[]')"

for build in $BUILDS
do
	echo "Build: $build"
	echo -e "\t$(curl -s ${IMAGE_URL_PREFIX}/${build}/meta.json | jq -r '.images.openstack.path')"

**# ./checkBuilds.sh https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/**
Build: 410.8.20190412.2
	rhcos-410.8.20190412.2-openstack.qcow2
Build: 410.8.20190412.1
	rhcos-410.8.20190412.1-openstack.qcow2
Build: 410.8.20190412.0
	rhcos-410.8.20190412.0-openstack.qcow2
Build: 410.8.20190411.1
	rhcos-410.8.20190411.1-openstack.qcow2
Build: 410.8.20190411.0
	rhcos-410.8.20190411.0-openstack.qcow2
Build: 410.8.20190410.0
	rhcos-410.8.20190410.0-openstack.qcow2
Build: 410.8.20190408.1
	rhcos-410.8.20190408.1-openstack.qcow2
Build: 410.8.20190408.0
	rhcos-410.8.20190408.0-openstack.qcow2
Build: 410.8.20190405.1
	rhcos-410.8.20190405.1-openstack.qcow2
Build: 410.8.20190405.0
	rhcos-410.8.20190405.0-openstack.qcow2
Build: 410.8.20190404.0
	rhcos-410.8.20190404.0-openstack.qcow2
Build: 410.8.20190322.0
	rhcos-410.8.20190322.0-openstack.qcow2
Build: 410.8.20190315.0
	rhcos-410.8.20190315.0-openstack.qcow2
Build: 410.8.20190314.0
	rhcos-410.8.20190314.0-openstack.qcow2

**# ./checkBuilds.sh https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa**
Build: 410.8.20190412.3
	null
Build: 410.8.20190412.2
	rhcos-410.8.20190412.2-openstack.qcow2
Build: 410.8.20190412.1
	null
Build: 410.8.20190412.0
	rhcos-410.8.20190412.0-openstack.qcow2
Build: 410.8.20190410.0
	rhcos-410.8.20190410.0-openstack.qcow2
Build: 410.8.20190405.0
	rhcos-410.8.20190405.0-openstack.qcow2
Build: 410.8.20190404.0
	rhcos-410.8.20190404.0-openstack.qcow2
Build: 410.8.20190402.0
	rhcos-410.8.20190402.0-openstack.qcow2
Build: 410.8.20190401.0
	rhcos-410.8.20190401.0-openstack.qcow2
Build: 410.8.20190330.0
	rhcos-410.8.20190330.0-openstack.qcow2
Build: 410.8.20190328.1
	rhcos-410.8.20190328.1-openstack.qcow2
Build: 410.8.20190328.0
	rhcos-410.8.20190328.0-openstack.qcow2
Build: 410.8.20190327.1
	rhcos-410.8.20190327.1-openstack.qcow2
Build: 410.8.20190327.0
	rhcos-410.8.20190327.0-openstack.qcow2
Build: 410.8.20190326.0
	rhcos-410.8.20190326.0-openstack.qcow2
Build: 410.8.20190325.0
	rhcos-410.8.20190325.0-openstack.qcow2
Build: 410.8.20190322.0
	rhcos-410.8.20190322.0-openstack.qcow2
Build: 410.8.20190320.1
	rhcos-410.8.20190320.1-openstack.qcow2
Build: 410.8.20190320.0
	rhcos-410.8.20190320.0-openstack.qcow2
Build: 410.8.20190319.0
	rhcos-410.8.20190319.0-openstack.qcow2
Build: 410.8.20190315.1
	rhcos-410.8.20190315.1-openstack.qcow2
Build: 410.8.20190315.0
	rhcos-410.8.20190315.0-openstack.qcow2
Build: 410.8.20190314.0
	rhcos-410.8.20190314.0-openstack.qcow2
Build: 410.8.20190313.0
	rhcos-410.8.20190313.0-openstack.qcow2

**Comment by gklein** [comment]
> Looking at [openshift/installer@8599867](https://github.com/openshift/installer/commit/8599867bef65c539e7fe2ea39dd1d1c5b4df79f5) it seems openshift/installer moved to a new location, and it seems like authentication is needed to access, currently looking at how we can align with that.

I don't think we need to authenticate in order to get an image. It is publicly available if we got the version right:

# curl -I -L "https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/ootpa/410.8.20190412.2/rhcos-410.8.20190412.2-openstack.qcow2"
HTTP/1.1 302 Moved Temporarily
Server: nginx/1.12.1
Date: Mon, 15 Apr 2019 12:12:57 GMT
Content-Type: text/html
Content-Length: 161
Location: https://art-rhcos-ci.s3.amazonaws.com/releases/ootpa/410.8.20190412.2/rhcos-410.8.20190412.2-openstack.qcow2
Set-Cookie: e864b3c9d0a263df2fa2591326969fb0=46b51d3581ea839913cb0632f75480f8; path=/; HttpOnly; Secure

HTTP/1.1 200 OK
x-amz-id-2: 9iuQW6DyC/NidECVxbg3n49ZeJHrFYS6z4N1+eFnhA9P6r3FUtwIWwwcSRXdRErWySgqLHQgIsc=
x-amz-request-id: FC9DB601BB21AA1D
Date: Mon, 15 Apr 2019 12:12:58 GMT
Last-Modified: Fri, 12 Apr 2019 20:08:51 GMT
ETag: "92bb79409af103b62856eb0267044b4d-92"
Content-Disposition: inline; filename=rhcos-410.8.20190412.2-openstack.qcow2
Content-Encoding: gzip
Accept-Ranges: bytes
Content-Type: binary/octet-stream
Content-Length: 769156524
Server: AmazonS3

**Comment by russellb** [comment]
Fixed by #359
